### PR TITLE
Update errorprone, google cloud java, grpc versions.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: gradle/errorprone/warnings [hunk 1]
```diff
@@ -62,7 +62,6 @@ errorProneWarnings = \
 -Xep:ReferenceEquality:ERROR,\
 -Xep:RequiredModifiers:ERROR,\
 -Xep:ShortCircuitBoolean:ERROR,\
--Xep:SimpleDateFormatConstant:ERROR,\
```
Comment (songy23) [inline gradle/errorprone/warnings]: Why remove this?

Comment (sebright) [inline gradle/errorprone/warnings]: Does this release add any warnings?

Comment (bogdandrutu) [reply gradle/errorprone/warnings]: It was removed:
> Task :opencensus-api:compileJava FAILED
com.google.errorprone.InvalidCommandLineOptionException: SimpleDateFormatConstant is not a valid checker name

Comment (bogdandrutu) [reply gradle/errorprone/warnings]: Added more new warnings.

**Pull request merged**